src/components/card/horizontal-video-card/horizontal-video-card.jsx

```diff
@@ -0,0 +1,64 @@
+import "./horizontal-video-card.css"
+import { useVideo } from "contexts/video-context";
+import { useState, useEffect, useRef } from "react"
+import { Link, useNavigate } from "react-router-dom";
+
+const HorizontalVideoCard = ({videoDetails}) => {
+    const {videoClickHandler} = useVideo();
+    const [dropdown,setDropdown] = useState(false)
+    const {_id,title,creator,thumbnail,views,uploadTime,duration,embedId} = videoDetails
+    const navigate = useNavigate();
+
+    const dropdownOptions = [
+        {optionText:"Add to watch later",optionIcon:"clock"},
+        {optionText:"Add to Playlist",optionIcon:"circle-play"},
+        {optionText:"Remove from watch later",optionIcon:"trash"}]
+    
+    const settingRef = useRef()
+    useEffect(()=>{
+        document.addEventListener("mousedown",(e)=>{
+            if(!settingRef.current?.contains(e.target)){
```

Comment:
Don't use `addEventListener` in react, Instead, just provide a listener when the element is initially rendered.. checkout https://reactjs.org/docs/handling-events.html

Reply:
Thanks for mentioning this. I will make the changes later.